### PR TITLE
Replace xml with Compose UI

### PR DESCRIPTION
**Pull request by LoryMK**
Signed-off-by: kucheria <ext-mariia.kucheriava@here.com>

### CODE REVIEW

File: kotlin-sample/build.gradle [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
     repositories {
         google()
         jcenter()
-        
+
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:7.2.2'
-        classpath "org.jetbrains.kotlin:kotlin-gradle-plugin:$kotlin_version"
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.7.10'
```
Comment (NazarKacharaba) [inline kotlin-sample/build.gradle:16]: use `kotlin_version`

File: kotlin-sample/app/src/main/res/layout/fragment_map.xml [hunk 2]
```diff
@@ -0,0 +1,12 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/res/layout/fragment_map.xml:2]: Why do we need `LinearLayout`, can we just use `ComposeView` as the root view?

File: kotlin-sample/build.gradle (hunk 1 above)
Comment (NazarKacharaba) [inline kotlin-sample/build.gradle:16]: use `kotlin_version`

File: kotlin-sample/app/src/main/java/com/here/android/example/kotlin/ui/theme/Type.kt [hunk 3]
```diff
@@ -0,0 +1,28 @@
...
+        fontFamily = FontFamily.Default,
+        fontWeight = FontWeight.W500,
+        fontSize = 14.sp
+    ),
+    caption = TextStyle(
+        fontFamily = FontFamily.Default,
+        fontWeight = FontWeight.Normal,
+        fontSize = 12.sp
+    )
+    */
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/java/com/here/android/example/kotlin/ui/theme/Type.kt:27]: remove commented code.

File: kotlin-sample/app/src/main/java/com/here/android/example/kotlin/ui/theme/Theme.kt [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+    secondary = Teal200
+
+    /* Other default colors to override
+    background = Color.White,
+    surface = Color.White,
+    onPrimary = Color.White,
+    onSecondary = Color.Black,
+    onBackground = Color.Black,
+    onSurface = Color.Black,
+    */
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/java/com/here/android/example/kotlin/ui/theme/Theme.kt:27]: remove commented code.

File: kotlin-sample/app/src/main/java/com/here/android/example/kotlin/ui/theme/Color.kt [hunk 5]
```diff
@@ -0,0 +1,8 @@
+package com.here.android.example.kotlin.ui.theme
+
+import androidx.compose.ui.graphics.Color
+
+val Purple200 = Color(0xFFBB86FC)
+val Purple500 = Color(0xFF6200EE)
+val Purple700 = Color(0xFF3700B3)
+val Teal200 = Color(0xFF03DAC5)
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/java/com/here/android/example/kotlin/ui/theme/Color.kt:8]: We also have identical colours defined in XML, why not use only one source?

File: kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt [hunk 6]
```diff
@@ -0,0 +1,303 @@
...
+                }
+            }
+            else {
+                NavigationManager.getInstance().also {navigationManager ->
+                    mapViewModel.setNavigationManager(navigationManager)
+                    navigationManager.setMap(map)
+                    navigationManager.mapUpdateMode = NavigationManager.MapUpdateMode.ROADVIEW_NOZOOM
+                    route?.let { navigationManager.simulate(it, 40) }
+                }
+            }
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt:218]: Use NavigationManager from one source(e.g. mapViewModel.getNavigationManager())

File: kotlin-sample/app/src/main/AndroidManifest.xml [hunk 7]
```diff
@@ -5,7 +5,8 @@
 
     <uses-permission android:name="android.permission.ACCESS_FINE_LOCATION" />
     <uses-permission android:name="android.permission.ACCESS_COARSE_LOCATION" />
-    <uses-permission android:name="android.permission.WRITE_EXTERNAL_STORAGE"
+    <uses-permission
+        android:name="android.permission.WRITE_EXTERNAL_STORAGE"
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/AndroidManifest.xml:9]: WRITE_EXTERNAL_STORAGE why do we need this permission?

File: kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt [hunk 8]
```diff
@@ -0,0 +1,303 @@
...
+                startSimulation()
+                ButtonTitles.FINISH_SIMULATION
+            }
+            ButtonTitles.FINISH_SIMULATION -> {
+                clearMapData()
+                ButtonTitles.CREATE_ROUTE
+            }
+        }
+
+        isRouteCalculated = buttonTitles == ButtonTitles.CREATE_ROUTE
```
Comment (NazarKacharaba) [inline kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt:247]: Integer ids( e.g. R.id.buttion_id) should be compared, not the title of the button.

**Review by NazarKacharaba — CHANGES_REQUESTED** [review]

File: kotlin-sample/app/src/main/res/layout/fragment_map.xml (hunk 2 above)
Comment (LoryMK) [reply kotlin-sample/app/src/main/res/layout/fragment_map.xml:2]: We can use ComposeView as the root view

File: kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt (hunk 8 above)
Comment (LoryMK) [reply kotlin-sample/app/src/main/java/com/here/android/example/kotlin/AppMapView.kt:247]: But I have one button with few titles and this button don't need to have an id

**Review by NazarKacharaba — APPROVED** [review]

**Pull request merged**